File: plaud-template-app/ios/PlaudTemplateApp/Models/RecordingFile.swift
import Foundation

/// Local recording file record
struct RecordingFile: Identifiable, Codable {
    let id: String                    // Local unique ID (UUID)
    let sessionId: Int                // Device-side sessionId (recording start timestamp)
    let deviceSN: String              // Source device SN
    var name: String                  // Display name (default "Untitled Recording")
    var duration: TimeInterval        // Duration in seconds (parsed from audio file, 0 before sync completes)
    let createdAt: Date               // Recording time (derived from sessionId)
    var syncedAt: Date?               // Sync completion time
    var localPath: String?            // Local audio path (available after sync)
    var summaryText: String?          // AI summary
    var transcriptJSON: String?       // Transcription data JSON

    var isSynced: Bool { localPath != nil }
}

extension RecordingFile {
    static let defaultName = "Untitled Recording"
}

File: android/app/src/main/java/com/plaud/template/models/RecordingFile.kt
package com.plaud.template.models

import com.google.gson.annotations.SerializedName
import java.util.UUID

data class RecordingFile(
    @SerializedName("id")
    val id: String = UUID.randomUUID().toString(),

    @SerializedName("sessionId")
    val sessionId: Long,

    @SerializedName("deviceSN")
    val deviceSN: String,

    @SerializedName("name")
    var name: String,

    @SerializedName("duration")
    var duration: Long,

    @SerializedName("createdAt")
    val createdAt: Long,

    @SerializedName("syncedAt")
    var syncedAt: Long? = null,

    @SerializedName("localPath")
    var localPath: String? = null,

    @SerializedName("summaryText")
    var summaryText: String? = null,

    @SerializedName("transcriptJSON")
    var transcriptJSON: String? = null
) {
    val isSynced: Boolean
        get() = localPath != null
}
